Modules/Loadable/SubjectHierarchy/Widgets/Python/SubjectHierarchyPlugins/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+
+# Deploy abstract scripted plugin and init file that
+# imports all deployed scripted plugins deployed by modules
+set(SubjectHierarchyPlugins_PYTHON_SCRIPTS
+  __init__
+  AbstractScriptedSubjectHierarchyPlugin
+  parameterNodeWrapper/__init__
+  parameterNodeWrapper/guiConnectors
```

Comment:
Since `ParameterNodeWrapperPlugins` is a shared folder name in the install tree (the same way as all other Python plugin folders), the filenames in it must be unique. Therefore `guiConnectors.py` is not a valid filename, it has to be `SubjectHierarchyGuiConnectors.py`.  Again, the naming convention was violated with guiConnectors.py (started with lowercase). `SubjectHierarchyGuiConnectors.py` is not perfect, because GUI should be all caps, but that would require changing of the class name as well, so I would not do it in this commit.

Reply:
I would argue that the approach to register the `parameterNodeWrapper` is more robust as the registration does not require the files to be physically located in the same folder to ensure proper registration. It is done through the `parameterNodeGuiConnector` decorator.

[^1]: https://github.com/Slicer/Slicer/blob/ff58cc7a4e013bb6f46bd867b6d09f920770be63/Base/Python/slicer/parameterNodeWrapper/guiConnectors.py#L117-L152

The current mechanism for `SubjectHierarchyPlugins` is brittle and it may be sensible to also adopt a decorator based registration.

> re: `gui` vs `GUI` naming

This was discussed a while back and if I recall, similarly to the fact we have `vtkSomething`, `qMRMLSomething`, `mrml`, ... it started with `gui` lower cased. I don't think we should change this.

Reply:
ParameterNodeWrapper stuff does not belong into SH plugins folder. It could go to a (moduleName)Lib folder if we don't want a shared folder. I would use a shared folder for consistency with other plugins. If we chose to use (moduleName)Lib folder for plugins instead of a shared folder (which could make sense) then we should do it for all plugins, such as SegmentEditor, SubjectHierarchy, SegmentStatistics, plugins.

Reply:
To follow-up on this discussion, the registration of parameterNodeWrapper plugins has been improved:
* https://github.com/Slicer/Slicer/pull/7480